lisa/util/parallel.py

```diff
@@ -2,29 +2,66 @@
 # Licensed under the MIT license.
 
 from concurrent.futures import FIRST_COMPLETED, Future, ThreadPoolExecutor, wait
-from typing import Any, Callable, Generic, List, Optional, TypeVar
+from typing import Any, Callable, Dict, Generic, List, Optional, TypeVar
+
+from lisa.util.logger import Logger, get_logger
+from lisa.util.perf_timer import create_timer
 
 from . import LisaException
 
 T_RESULT = TypeVar("T_RESULT")
 
 
+class Task(Generic[T_RESULT]):
+    def __init__(
+        self,
+        task_id: int,
+        task: Callable[[], T_RESULT],
+        parent_logger: Optional[Logger],
+    ) -> None:
+        self.id = task_id
+        self._task = task
+        self._log = get_logger("Task", str(self.id), parent_logger)
+        self._log.debug(f"Generate task: {self}")
+
+    def close(self) -> None:
```

Comment:
After moved the line of log, it looks the close method is not necessary.

Reply:
We still need close method to log the time taken from starting the task to when it joined with the root thread. This would be helpful to debug if any task is taking a long time.